pkg/asset/cluster/azure/azure.go

```diff
@@ -12,3 +20,42 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 		Region: config.Platform.Azure.Region,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	clusterName := installConfig.Config.ObjectMeta.Name
+	azureDNS, _ := azureconfig.NewDNSConfig()
+	record := fmt.Sprintf("api.%s", clusterName)
+
+	exists := false
+
+	// Look for an existing CNAME first
+	rs, err := azureDNS.GetDNSRecordSet(installConfig.Config.Platform.Azure.PublicZoneID, record, "CNAME")
```

Comment:
Nit: use `azdns.CNAME`, `azdns.A`, `azdns.AAAA`, instead of strings?

Reply:
Yes. This was supposed to happen but slipped. Fixed!